### DISCUSSION TITLE
6022BE fails to connect for firmware upload

### DISCUSSION THREAD

**Discussion by Fuergrissa1** [OpenHantek/OpenHantek6022#357] [Q&A] [answered]
I have OpenHantek6022 v3.3.2_armhf.deb installed on raspbian bullseye (raspi 4b). The software detects my 6022BE which is the only device plugged into USB, but when selected on the list a screen showing a connection error flashes very quickly and the application doesn't start (in measurement mode), only giving the option to enter demo mode. I cannot read the details because the screen flashes so quickly, but it seems to say something like "Unable to connect....Permission denied".

Could I have some advice on how to troubleshoot this? The scope does work on my windows machine with the OEM software.

**Comment by Ho-Ro** [comment, marked as answer]
You have a rights problem. 
You could start the program from a terminal and look for error messages. Increase the verbosity with `OpenHantek --verbose 3`, this gives on my system (only the first relevant lines):
```
Version: 3.3.2.1-32-gec5f7c9 - build: Mar  7 2023 19:18:56
0 ms: create openHantekApplication
145 ms: set "Fusion" style, system style is "breeze"
145 ms: set RT FIFO scheduler
145 ms: load translations for locale "C"
145 ms: init libusb
153 ms: show splash screen
 FindDevices::FindDevices()
  FindDevices::updateDeviceList()
  +++ "0x2100000004b40000" "DSO-6022BE"
  DevicesListModel::updateDeviceList() "DSO-6022BE" upload firmware
  FindDevices::updateDeviceList()
  --- "0x2100000004b40000" "DSO-6022BE"
  FindDevices::updateDeviceList()
  +++ "0x2100000004b50210" "DSO-6022BE"
  DevicesListModel::updateDeviceList() "DSO-6022BE" can connect
2189 ms: use device "DSO-6022BE" serial number "C3B36055284A"
2189 ms: create DSO control thread
...
```
The lines with `"0x2100000004b40000" "DSO-6022BE"` locate the device on USB, in my case `/dev/bus/usb2` (the first eigth digits show the usb tree, in my case the 1st device on usb2, while the last 8 digits show the VID (`0x04b4` = unconfigured Hantek) and the FW version (`0x0000` = no FW loaded). After loading the FW it changes to `"0x2100000004b50210" "DSO-6022BE"`, same bus position, but VID changed to `0x04b5` (configured Hantek) and FW 02.10 was loaded.
After plugging in (no need to run OpenHantek) you could check the USB rights with `ls -lR /dev/bus/usb`, this gives in my example (lot of lines deleted):
```
/dev/bus/usb:
total 0

...

/dev/bus/usb/002:
total 0
crw-rw-r--  1 root root    189, 128 Feb 25 18:48 001
crw-rw-rw-+ 1 root plugdev 189, 164 Mar  8 09:23 037

...
```

The last line is my device, that has rw access for the group `plugdev`
You should check if you're member of the group `plugdev` with the command `groups`, in my case:
```
... lp dialout cdrom floppy audio dip video plugdev users ...
```
If you're not member of this group you can add you to this group with `sudo adduser YOUR_USER_NAME plugdev`
